frontend/src/modules/WellCompletion/wellCompletionsDataAccessor.ts

```diff
@@ -0,0 +1,236 @@
+export type TimeAggregation = keyof typeof TimeAggregations;
+
+export class WellCompletionsDataAccessor {
+    /**
+     * This class is responsible for parsing and providing data for the well completions plot.
+     *
+     * It parses data structure received from backend and creates the data structure needed for the plot
+     * in WellCompletionsPlot-component.
+     *
+     * The data is parsed from the WellCompletionData_api object and methods are taken from the
+     * WellCompletions-component of webviz-subsurface-components package, and modified to fit.
+     */
+    private _dataSet: WellCompletionDataSet_api | undefined;
+    private _subzones: WellCompletionZone_api[];
+    private _wells: WellCompletionWell_api[];
+    private _searchWellText: string;
+    private _hideZeroCompletions: boolean;
+
+    constructor() {
+        this._dataSet = undefined;
+        this._subzones = [];
+        this._wells = [];
+        this._searchWellText = "";
+        this._hideZeroCompletions = false;
+    }
+
+    parseWellCompletionsData(data: WellCompletionData_api): void {
+        // TODO:
+        // - Filter wells when filter functionality is in place
+        // - Filter subzones when filter functionality is in place
+
+        this._dataSet = data.json_data;
+        this._wells = this._dataSet.wells;
+
+        // Exctract all subzones
+        this._subzones = [];
+        this._dataSet.stratigraphy.forEach((zone) => this.findSubzones(zone, this._subzones));
+    }
+
+    setNumberOfWellsToShow(numberOfWellsToShow: number): void {
+        // TODO: Add functionality for pagination or similar to control number of wells to include in plotData.
+        void numberOfWellsToShow;
+        return;
+    }
+
+    setSearchWellText(searchWell: string): void {
+        this._searchWellText = searchWell;
+    }
+
+    setHideZeroCompletions(hideZeroCompletions: boolean): void {
+        this._hideZeroCompletions = hideZeroCompletions;
+    }
+
+    getTimeSteps(): string[] {
+        if (this._dataSet === undefined) return [];
+        return this._dataSet.timeSteps;
+    }
+
+    createPlotData(timeStepSelection: string | [string, string], timeAggregation: TimeAggregation): PlotData | null {
+        // TODO: Consider removing function arguments, and use setter-methods for each argument and set to an attribute.
+        //       This would make it easier to modify/adjust single attributes and call "createPlotData" again.
+        if (!this._dataSet) return null;
+
+        let timeStepIndexRange: [number, number] | null = null;
+        if (typeof timeStepSelection === "string") {
+            const timeStepIndex = this._dataSet ? this._dataSet.timeSteps.indexOf(timeStepSelection) : 0;
+            timeStepIndexRange = [timeStepIndex, timeStepIndex];
+        } else {
+            timeStepIndexRange = [
+                this._dataSet.timeSteps.indexOf(timeStepSelection[0]),
+                this._dataSet.timeSteps.indexOf(timeStepSelection[1]),
```

Comment:
Be aware that `indexOf` returns `-1` if nothing was found. Maybe use 0 in that case?

Reply:
Keeping the -1 and handle check with `if (timeStepIndexRange[0] === -1 || timeStepIndexRange[1] === -1) return null;` after setting of `timeStepIndexRange`. This makes sure no created plot data to be inconsistent with selected time steps in settings